analyses/survival-analysis/survival-analysis_HGG_DMG.Rmd

```diff
@@ -0,0 +1,210 @@
+  select(Kids_First_Participant_ID ,molecular_subtype) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID ) %>% 
+  summarise(count=n(),subtype=toString(molecular_subtype)) %>%
+  filter(count>1)
+
+# these patients might affect downstream analysis for survival and would need to be removed
+count_patient_subtype
+
+```
+
+#### Let's use independent_primary to use primary samples only and maybe will help filter cases where multiple subtypes are present per patient ID
+
+```{r Read in independent}
+independent_primary<-read_tsv(independent_primary_file)
+# select only independent primary bs_ids to get overall survival
+metadata<-metadata %>% filter(Kids_First_Biospecimen_ID %in% independent_primary$Kids_First_Biospecimen_ID)
+
+# find patients where multiple subtypes are present 
+count_patient_subtype<-metadata %>% 
+  select(Kids_First_Participant_ID ,molecular_subtype) %>%
+  unique() %>%
+  group_by(Kids_First_Participant_ID ) %>% 
+  summarise(count=n(),subtype=toString(molecular_subtype)) %>%
+  filter(count>1)
+
+count_patient_subtype
+
+```
+#### Ok so no patients with multiple subtypes now.
+
+## Kaplan-Meier for all HGG/DMG subtypes
+
+The [Kaplan-Meier](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059453/) 
+survival curve shows the probability of survival over time.
+
+```{r Kaplan Meier}
+kap_fit <- survival_analysis(metadata,
+  ind_var = "molecular_subtype",
+  test = "kap.meier",
+  metadata_sample_col = "Kids_First_Biospecimen_ID"
+)
+# median and 0.95CI survival days
+kap_fit$model
+```
+
+`survminer` package comes with a convenient plotting function which uses `ggplot2`
+arguments. 
+It requires the original Kaplain-Meier `survfit` object, so we will extract that with `$model`.
+
+```{r Make survival plot}
+surv_plot <- survminer::ggsurvplot(kap_fit$model,
+  pval = TRUE,
+  data = kap_fit$original_data,
+  risk.table = TRUE,
+  xlim = c(0, 3000),
+  break.time.by = 500,
+  ggtheme = theme_minimal(),
+  risk.table.y.text.col = TRUE,
+  risk.table.y.text = FALSE
+) 
+
+surv_plot$plot<-surv_plot$plot+ggtitle(paste0("Kap_meier_Diffuse_astrocytic_and_oligodendroglial_tumor"))+theme(legend.position = "right")
+
+# Make this plot a combined plot
+surv_plot_all_subtype <- cowplot::plot_grid(surv_plot[[1]], surv_plot[[2]], nrow = 2, 
+                                rel_heights = c(3, 2))
+
+# Print it out here
+surv_plot_all_subtype
+```
+
+Save the plot to a file. 
+
+```{r Save survival plot}
+# We can save the plot like a normal ggplot
+cowplot::save_plot(filename = kap_meier_plot_file, plot = surv_plot_all_subtype)
+# Let's save the model for CI and median survial as well
+write_tsv(surv_median(kap_fit$model),kap_meier_model_file)
+```
+
+### Subtypes comapred to wildtype
+Lets look at Kap Meier plots for each mutant subtype to H3 wildtype
+
+```{r Kaplan Meier per subtype compared to wildtype}
+
+list_subtypes<-c("DMG, H3 K28","BRAF V600E","HGG, H3 G35","HGG, IDH","DMG, H3 K28, BRAF V600E")
```

Comment:
This strikes me as a potentially concerning amount of multiple testing, but I'm still not the expert to ask. @jaclyn-taroni let's find one!